File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.js
function solution(arr) {
    let answer = arr.slice();
    let count = 0;

    while (true) {
        for (let i = 0; i < arr.length; i++) {
            if (arr[i] >= 50 && arr[i] % 2 === 0) {
                arr[i] /= 2;
            } else if (arr[i] < 50 && arr[i] % 2 !== 0) {
                arr[i] = arr[i] * 2 + 1;
            }
        }

        if (JSON.stringify(arr) === JSON.stringify(answer)) {
            break;
        }
        answer = arr.slice();

        count++;
    }
    return count;
}

File: 프로그래머스/0/181881. 조건에 맞게 수열 변환하기 2/조건에 맞게 수열 변환하기 2.py
def solution(arr):
    prev_arr_state = list(arr) 
    count = 0
    while True:
        current_arr_state = []
        changed = False
        for num in prev_arr_state:
            if num >= 50 and num % 2 == 0:
                new_num = num // 2
            elif num < 50 and num % 2 != 0:
                new_num = num * 2 + 1
            else:
                new_num = num
            current_arr_state.append(new_num)
            if new_num != num:
                changed = True
        if not changed:
            break
        prev_arr_state = current_arr_state
        count += 1
    return count